### DISCUSSION TITLE
Does Duckdb suffers from "Database file is locked" like SQLite?

### DISCUSSION THREAD

**Discussion by alexe100** [duckdb/duckdb#8126] [General]
Does Duckdb suffers from "Database file is locked" like SQLite?
Any chance of happening?

Thanks a lot
Imo

**Comment by Alex-Monahan** [comment]
Hello! Does our [FAQ](https://duckdb.org/faq) help answer your question?

**Comment by alexe100** [comment]
I read it but it is not clear for me. Is there the chance of the application crash and let database file locked?
Does Duckdb uses file system locking or it uses in-memory semaphores and mutexes that do not let db file locked ("for ever")?

Thank you

**Reply by Tishj** [reply]
I believe we use mutexes, precisely I think we use this file and the classes in it
`src/include/duckdb/storage/storage_lock.hpp`

**Comment by bladko-tom** [comment]
I am trying to use SQlite Scanner to update and insert into Sqlite DB with duckDB just in memory DB + Attach SQLite DB to process it, but all the the time i am getting database is locked. Its really hard to provide steps to duplicate so far it looks like random 

```
DB = duckdb.connect()
DB.execute("ATTACH 'SLITE.db' AS SQLITEDB (TYPE SQLITE, READ_ONLY FALSE)")
DB.execute("USE SQLITEDB")
DB.execute("DROP TABLE IF EXISTS AAA")
DB.execute("CREATE TABLE AAA AS SELECT 10 AS FIELD, True as TGB;")
DB.execute("UPDATE AAA SET FIELD = 20")
DB.execute("INSERT INTO AAA BY NAME SELECT 30 AS FIELD, False as TGB;")
DB.execute("UPDATE AAA SET FIELD = 10 WHERE TGB = 1")
DB.execute("INSERT INTO AAA BY NAME SELECT 30 AS FIELD, False as TGB;")
DB.execute("use memory")
DB.execute("DETACH SQLITEDB")
DB.close()
```
